docs/guide/index.asciidoc

```diff
@@ -68,6 +68,20 @@ open source Kibana dashboards that come with APM Server.
 We designed Elastic APM to run on anything from a regular laptop to thousands of machines,
 and it's easy to get started.
 
+[[separate-component]]
+== Why is APM Server a separate component?
+
+The APM Server is kept as a separate component for the following reasons:
+
+* Agents can offload their heavy lifting tasks to the APM Server which can be scaled independently,
+  since it is a separate component thereby reducing the load on your app and by keeping the agents as light as possible.
+* During real-user-monitoring when the data is being collected in the browsers agents talk to the independent APM Server which acts as a central point to control the amount of data flowing into Elasticsearch and also control the pace of ingesting data.
+  This also prevents the browser from directly interacting with Elasticsearch which is not recommended.
+* In cases where Elasticsearch is temporarily down we can't rely on the agents to buffer up the data since that increases the memory footprint of the application.
+  APM Server does this buffering since we can't rely on disk access being available to the agents.
+* It serves as a middleware for source mapping for javascript in the browser.
+* The APM Server provides a JSON API for agents to use thereby guaranteeing better maintain compatibility across different versions of agents and Elasticsearch.
```

Comment:
The APM Server provides a JSON API for agents to use thereby improving compatibility across different versions of agents and Elasticsearch.